### ISSUE TITLE
Need indicator to user to differentiate between user-uploaded and Expression Database data

### ISSUE THREAD

**Issue by kdahlquist** [dondi/GRNsight#863]
I see how the functionality works now.  The user uploaded data and Expression Database data are both accessible from the drop-down menus, which is what we wanted.

However, there is no indicator to the user that this is the case.  Also, our Expression Database is only for yeast right now.

We need to provide some more info to the user about this.  I'm pretty braindead right now so I don't have any suggestions off the top of my head, but I wanted to note this as an issue.

**Comment by dondi** [comment]
Possibly prepending text or looking into `optgroup` may make this low-hanging fruit

**Comment by ntran18** [comment]
I think we need to *sit down* and think where to put it before I can implement anything 😅.

**Comment by kdahlquist** [comment]
There would be two optgroups:
- User-uploaded
- Expression Database

**Comment by MilkaZek** [comment]
We went through the various pages and identified this section (referring to the attached screenshot/code snippet) of the update-app.js file to be where we need to separate the drop-down menu options - we (@Amelie1253 and I) plan to attempt that fix this upcoming week after clarifying which list options fall under the category of User-uploaded or Expression Database.

<img width="741" alt="Image" src="https://github.com/user-attachments/assets/5c106bd3-83fb-4d4c-a453-014f7f437714" />

**Comment by dondi** [comment]
In response to this comment…

> We went through the various pages and identified this section (referring to the attached screenshot/code snippet) of the update-app.js file to be where we need to separate the drop-down menu options - we ([@Amelie1253](https://github.com/Amelie1253) and I) plan to attempt that fix this upcoming week after clarifying which list options fall under the category of User-uploaded or Expression Database.

The code snapshot points to the right place; the values listed in line 714 should be what eventually gets wrapped in an `opt-group` labeled “User-uploaded” and the line 721 block is what gets wrapped in an “Expression database” `opt-group`

**Comment by dondi** [comment]
For question regarding the long comment, that pertains to #865 —we have backlogged it because the workaround appears to remain stable and we haven’t had the bandwidth to investigate this further for a “real” solution

We can help future developers by cross-linking between this comment and the issue (i.e., mention the issue number in the comment, and perhaps in the issue, mention the file/approximate location of the workaround)

**Comment by kdahlquist** [comment]
Upon testing, we realized that we needed clarification of the behavior:
- Currently, the list of datasets are not clearing between selecting new graphs (either from demo, user-upload, or database).  They will need to clear in between.  I recall that we saw this bug before a long time ago.  I'm sure there is a closed issue about it lurking somewhere.
- We need to add a third category to the `opt-group` called "Demo".  This is because a demo file is not really a user-uploaded file, it is a category unto itself.
- When a user loads a graph from the database, the demo and user-uploaded categories would not be filled.  Only the expression database category would have things to choose from.

**Comment by MilkaZek** [comment]
The clearing issue has now been fixed, but we have some questions and need clarification on what belongs in each of the three categories as well as what is meant by the third bullet point from the previous comment.

**Comment by dondi** [comment]
The category question was clarified by @kdahlquist; the “User-Uploaded” opt-group needs to be called “Demo” if the source of the current network was a demo. “User-Uploaded” should only be used if the current graph was loaded from a file (“Open File…” option in either the sidebar or the menu bar)

**Comment by MilkaZek** [comment]
We may have found a new bug - as we were playing around with the demos and user-uploaded feature, we ran into an issue where when we selected a file to upload, the viewport would not change and show the network we uploaded, staying stuck on what was there previously. I could not identify a pattern to this yet, just that most of the time after clicking on a demo or generating a network, trying to upload a file (we used an exported file of Demo 1) did not result in the correct response.  We have a screen recording of the issue that we can show during our 4/9 meeting.

**Comment by Amelie1253** [comment]
I’ve come up a solution for the original issue for differentiating dropdown selections into the optgroups based on how the data was uploaded. This was done by adding a new grnState variable to track the upload route and label the optgroups accordingly. I've pushed the changes to the amelie-milka-863 branch. However, I accidentally committed some extra files and haven't fully tested the functionality, so I wasn't sure if if I should open a pull request yet.

**Comment by dondi** [comment]
As a next step in investigating @MilkaZek’s reported bug from https://github.com/dondi/GRNsight/issues/863#issuecomment-2788281396, we should identify/isolate exactly which version or deployment of GRNsight has this behavior:
* Can it be replicated on the deployed `beta`?
* Can it be replicated on an untouched `beta` branch local copy?
* Can it be replicated, similarly, on the `main` branch? (deployed and/or local)
* Might it be isolated to the work-in-progress branch for this original issue?

If we are able to replicate this consistently, we should it report it as a new issue

**Comment by dondi** [comment]
Back to [this comment](https://github.com/dondi/GRNsight/issues/863#issuecomment-2788485512) on the original issue, yes just clean up and test as needed, then open a PR whenever you feel it is ready

**Comment by MilkaZek** [comment]
> As a next step in investigating [@MilkaZek](https://github.com/MilkaZek)’s reported bug from [#863 (comment)](https://github.com/dondi/GRNsight/issues/863#issuecomment-2788281396), we should identify/isolate exactly which version or deployment of GRNsight has this behavior:
> 
> * Can it be replicated on the deployed `beta`?
> * Can it be replicated on an untouched `beta` branch local copy?
> * Can it be replicated, similarly, on the `main` branch? (deployed and/or local)
> * Might it be isolated to the work-in-progress branch for this original issue?
> 
> If we are able to replicate this consistently, we should it report it as a new issue

We tested for this bug on the deployed `beta` and local main, and found similar results to our work-in-progress branch for this original issue: when we seem to go in order of demos (click Demo #1, upload custom demo, click Demo #2, upload custom demo, etc.) it tends not to work and the viewport doesn't change, but it's slightly inconsistent, especially when I click demos out of order to test custom uploading. Based on what we tried, this behavior does not seem to be isolated to our branch for this original issue.

**Comment by kdahlquist** [comment]
I see that this functionality is now in beta 7.2.6.  Upon testing, I have two comments about how it looks, and then some bugs to report that may or may not be a result of working with this new functionality.  I'm going to leave the comments on the look here and begin a new issue for the bugs so that the issues can be separated.

- For "User-uploaded", the second "u" needs to be lower case. 
- Will you please move the "User-uploaded" and "Expression Database" labels to the left margin instead of being centered?

**Comment by dondi** [comment]
For the second bullet, it appears that the default is browser-dependent but it might be overridable with the right CSS; quick experiment during the meeting didn’t appear to work but a little more investigation may be worthwhile

May be worth a brief “research spike”

**Comment by Amelie1253** [comment]
I found [this stack overflow](https://stackoverflow.com/questions/7452479/how-to-change-the-style-of-a-selects-optgroup-label) that discusses opt-group styling. I mostly skimmed it and it seems like opt-groups are particularly hard to style with CSS as the webpage overrides the styling. There are some recommended methods on how to circumvent that issues and I will try out these methods in a bit.

**Comment by MilkaZek** [comment]
Just to confirm, are we wanting to try or move away from overriding the browser default styling for the opt-groups?

**Comment by dondi** [comment]
Yes, we will no longer pursue the opt-group styling part. Sorry just realized we didn’t transcribe that decision as a note 😅

So I think this ticket is now solely about making sure that those lists empty out whenever something new is reloaded (plus the capitalization change)

**Comment by kdahlquist** [comment]
This is complete in beta v7.2.9--there is one outstanding comment in the related issue #1211.
